src/uproot/interpretation/library.py

```diff
@@ -955,130 +871,131 @@ def group(self, arrays, expression_context, how):
-                            # >>> from_index = pandas.MultiIndex.from_tuples([(0,), (1,)])
-                            # >>> to_index = pandas.MultiIndex.from_tuples([(0, 0), (0, 1), (0, 2), (1, 0), (1, 1)])
-                            # >>> pandas.Series([1.1, 4.4], index=from_index).reindex(to_index)
-                            # 0  0    1.1
-                            #    1    1.1
-                            #    2    1.1
-                            # 1  0    4.4
-                            #    1    4.4
-                            # dtype: float64
-                            df.append(
-                                pandas.Series(array.values, index=flat_index).reindex(
-                                    index
-                                )
-                            )
-                            gn.append(name)
-                        elif name in group:
-                            df.append(array)
-                            gn.append(name)
-                out = []
-                for index, df, gn in zip(indexes, dfs, group_names):
-                    out.append(
-                        pandas.DataFrame(
-                            data=dict(zip(gn, df)), index=index, columns=gn
-                        )
-                    )
-                if len(out) == 1:
-                    return out[0]
-                else:
-                    return tuple(out)
-            else:
-                out = None
-                for index, group in zip(indexes, groups):
-                    only = {name: arrays[name] for name in group}
-                    df = pandas.DataFrame(data=only, index=index, columns=group)
-                    if out is None:
-                        out = df
-                    else:
-                        out = pandas.merge(
-                            out, df, how=how, left_index=True, right_index=True
-                        )
-                flat_names = [
-                    name
-                    for name in names
-                    if _is_pandas_rangeindex(pandas, arrays[name].index)
-                ]
-                if len(flat_names) > 0:
-                    flat_index = pandas.MultiIndex.from_arrays(
-                        [arrays[flat_names[0]].index]
-                    )
-                    only = {
-                        name: pandas.Series(arrays[name].values, index=flat_index)
-                        for name in flat_names
-                    }
-                    df = pandas.DataFrame(
-                        data=only, index=flat_index, columns=flat_names
-                    )
-                    out = pandas.merge(
-                        df.reindex(out.index),
-                        out,
-                        how=how,
-                        left_index=True,
-                        right_index=True,
-                    )
-                return out
+            return _pandas_memory_efficient(pandas, arrays, names)
```

Comment:
Now `_pandas_memory_efficient` is the _only_ way DataFrames are made. Is it still necessary?

Calling `gc.collect()` is not a good thing to do in production code; I only acquiesced because a user was having real trouble with the intermediate memory use of building a whole DataFrame at once. I would _love_ to be able to justify removing `gc.collect()` from the codebase: that is, ensuring that we're combining Series that all have a common RangeIndex into a DataFrame without a lot of copying or shuffling. It seems to me like the DataFrame should just adopt the memory used by the Series. If there's some flag in Pandas to make that happen, it would be ideal. We know, for instance, that the original Series are not going to be used anymore; the new DataFrame does not need to copy the Series for safety.